commit f9cef2566ba5e2fc0732256eb726017a8820ca77
Author: David Woodhouse <David.Woodhouse@intel.com>
Date:   Thu Jun 24 15:41:30 2010 +0100

    Remove courier-imap rename workaround
    
    This is breaking things on non-buggy servers, because after renaming the
    folder in question we also attempt to rename all its subfolders... and
    quite reasonably get a failure.
    
    Testing against current versions of Courier (4.4.0) seems to suggest
    that it's not actually necessary, either -- perhaps the bug has been
    fixed in the 9 years since this hack was first added to the imap back
    end in commit bfce883e.
    (cherry picked from commit 4083bbc36154ea91b203d7b672df5584978dfb10)

diff --git a/camel/providers/imapx/camel-imapx-store.c b/camel/providers/imapx/camel-imapx-store.c
index 72ec38a..7556773 100644
--- a/camel/providers/imapx/camel-imapx-store.c
+++ b/camel/providers/imapx/camel-imapx-store.c
@@ -650,12 +650,6 @@ rename_folder_info (CamelIMAPXStore *istore, const gchar *old_name, const gchar
 				npath = g_strdup(new_name);
 			nfull = camel_imapx_store_summary_path_to_full(istore->summary, npath, istore->dir_sep);
 
-			/* workaround for broken server (courier uses '.') that doesn't rename
-			   subordinate folders as required by rfc 2060 */
-			if (istore->dir_sep == '.') {
-				camel_imapx_server_rename_folder (istore->server, path, nfull, ex);
-			}
-
 			camel_store_info_set_string((CamelStoreSummary *)istore->summary, si, CAMEL_STORE_INFO_PATH, npath);
 			camel_store_info_set_string((CamelStoreSummary *)istore->summary, si, CAMEL_IMAPX_STORE_INFO_FULL_NAME, nfull);
 
